cluster-service/deploy/integration/cluster-service-namespace.yaml

```diff
@@ -72,3 +80,66 @@ objects:
       namespace: ${NAMESPACE}
     data:
       cs-client-id: ${CLIENT_ID}
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: orphaned-namepsace-mgmt-cronjob
+      namespace: ${MGMT_ORPHANED_NAMESPACE}
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: orphaned-namespace-mgmt-cronjob
+    subjects:
+      - kind: ServiceAccount
+        name: orphaned-namepsace-mgmt-cronjob
+        namespace: ${MGMT_ORPHANED_NAMESPACE}
+    roleRef:
+      kind: ClusterRole
+      name: namespace-admin
+      apiGroup: rbac.authorization.k8s.io
+  - apiVersion: batch/v1
+    kind: CronJob
+    metadata:
+      name: orphaned-namespace-mgmt
+      namespace: ${MGMT_ORPHANED_NAMESPACE}
+    spec:
+      schedule: "0 0 * * *"
+      successfulJobsHistoryLimit: 3
+      failedJobsHistoryLimit: 1
+      jobTemplate:
+        spec:
+          template:
+            spec:
+              serviceAccountName: orphaned-namepsace-mgmt-cronjob
+              containers:
+                - name: kubectl-container
+                  image: ${KUBECTL-IMAGE}
+                  command: ["/bin/sh", "-c"]
+                  args:
+                    - |
+                      echo "Starting to clear orphaned namespaces"
+                      export CLUSTER_NAME="aro-hcp-aks"
+
+                      echo "setting cluster with server $KUBE_API_SERVER_HOST"
+                      kubectl config set-cluster $CLUSTER_NAME \
+                        --server=$KUBE_API_SERVER_HOST \
+                        --embed-certs \
+                        --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+
+                      echo "setting kubectl scredential"
+                      kubectl config set-credentials $SERVICE_ACCOUNT_NAME \
+                        --token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+                      kubectl config set-context current-context --cluster=$CLUSTER_NAME --user=$SERVICE_ACCOUNT_NAME
+                      kubectl config use-context current-context
```

Comment:
since the pod runs within the cluster, there is no need to set up a kubeconfig. just make sure that the job runs as the service account specified by $SERVICE_ACCOUNT_NAME